src/v/cluster/types.h

```diff
@@ -348,7 +348,9 @@ struct begin_tx_request : serde::envelope<begin_tx_request, serde::version<0>> {
     }
 };
 
-struct begin_tx_reply : serde::envelope<begin_tx_reply, serde::version<1>> {
+struct begin_tx_reply
+  : serde::
+      envelope<begin_tx_reply, serde::version<1>, serde::compat_version<0>> {
```

Comment:
Yes, tx are expected to work in a mixed mode cluster where an upgraded reader (v1) should be able to parse v0 reply from a partition. So this lgtm. The situation is a bit unlikely in practice though since transactions were disabled by default prior to 22.3 release.